optimade/server/config.py

```diff
@@ -0,0 +1,82 @@
+import json
+from configparser import ConfigParser
+from pathlib import Path
+
+
+class Config:
+    """Base class for loading config files and its parameters"""
+
+    ftype = "ini"
+
+    def __init__(self, ftype: str = None):
+        ftype = self.ftype if ftype is None else ftype
+        self.load(ftype)
+
+    def _get_load_func(self, format_name):
+        return getattr(self, f"load_from_{format_name}")
+
+    def load(self, ftype: str = None):
+        try:
+            f = self._get_load_func(ftype)
+        except AttributeError:
+            raise NotImplementedError(
+                f"load function for config format {ftype} is not implemented"
+            )
+        else:
+            f()
+
+
+class ServerConfig(Config):
+    """ This class stores server config parameters in a way that
+    can be easily extended for new config file types.
+
+    """
+
+    ftype = "ini"
```

Comment:
Is not really needed for this particular class, since its parent class defines this attribute similarly.